### PR TITLE
Migrate to poetry

### PR DESCRIPTION
**Pull request by yalef**
- Migrate all invokes to `saritasa-invocations`
- Drop support of python 3.9
- Add support for python 3.12
- Add django 5.0 to ci matrix
- Migrate from `pip-tools` to `poetry`
- Add configuration for  `coverage` tool
- Add separate invoke for `coverage` tests run

### CODE REVIEW

File: import_export_extensions/widgets.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def __init__(self, filename: str):
     def render(self, value: Model, *args, **kwargs) -> typing.Optional[str]:
         """Convert DB value to URL to file."""
         if not value:
-            return None
+            return None  # pragma: no cover
```
Comment (NikAzanov) [inline import_export_extensions/widgets.py:290]: Why empty values shouldn't be covered?

File: pyproject.toml [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-build-backend = "setuptools.build_meta"
-
-[project]
-authors = [
-    {name = "Saritasa", email = "pypi@saritasa.com"},
-]
-requires-python = ">=3.9"
+[tool.poetry]
+name = "django-import-export-extensions"
+version = "0.4.2"
```
Comment (NikAzanov) [inline pyproject.toml:3]: Current version is `0.4.1`, the next one will be `0.5.0` probably (or 1.0.0)

```suggestion
version = "0.4.1"
```

**Review by NikAzanov — COMMENTED** [review]
Please, check if we can get rid of `import_export_extensions/__init__.py` file content

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -12,7 +12,6 @@ jobs:
       fail-fast: false
       matrix:
         python-version:
-          - '3.9'
           - '3.10'
           - '3.11'
```
Comment (TheSuperiorStanislav) [inline .github/workflows/checks.yml:16]: Is it possible to add 3.12?

File: provision/project.py [hunk 4]
```diff
@@ -1,43 +1,19 @@
+import saritasa_invocations
 from invoke import task
 
-from . import common, django, docker, git, linters, tests
-
 
 ##############################################################################
 # Build project locally
 ##############################################################################
```
Comment (TheSuperiorStanislav) [inline provision/project.py:7]: Let's remove this

File: pyproject.toml [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+description = "Extend functionality of `django-import-export`"
+authors = ["Saritasa <pypi@saritasa.com>"]
+keywords = ["django_import_export_extensions"]
```
Comment (TheSuperiorStanislav) [inline pyproject.toml:6]: How about adding these keywords? 
image

File: pyproject.toml [hunk 6]
```diff
@@ -1,12 +1,12 @@
-[build-system]
-requires = ["setuptools>=61.0"]
-build-backend = "setuptools.build_meta"
-
-[project]
-authors = [
-    {name = "Saritasa", email = "pypi@saritasa.com"},
-]
-requires-python = ">=3.9"
+[tool.poetry]
```
Comment (TheSuperiorStanislav) [inline pyproject.toml:1]: Please add `maintainers`, `homepage`, `repository`

File: pyproject.toml [hunk 7]
```diff
@@ -22,23 +22,60 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
```
Comment (TheSuperiorStanislav) [inline pyproject.toml:23]: No need for that, poetry can set it by itself

File: pyproject.toml [hunk 8]
```diff
@@ -22,23 +22,60 @@ classifiers = [
...
+django-import-export = "^3.3.3"
+celery = "^5.3.6"
+redis = "^5.0.1"
+django = ">=3.2"
+djangorestframework = "^3.14.0"
+drf-spectacular = "^0.26.5"
+requests = "^2.31.0"
+django-filter = "^23.4"
+django-extensions = "^3.2.3"
+django-picklefield = "^3.1"
```
Comment (TheSuperiorStanislav) [inline pyproject.toml:37]: Let's add comments like we usually do

File: pyproject.toml [hunk 9]
```diff
@@ -22,23 +22,60 @@ classifiers = [
...
-license = {file = "LICENSE"}
-keywords = ["django_import_export_extensions"]
-name = "django_import_export_extensions"
-dynamic = ["dependencies", "version"]
 
-[project.urls]
-"Homepage" = "https://github.com/saritasa-nest/django-import-export-extensions"
-"Bug Tracker" = "https://github.com/saritasa-nest/django-import-export-extensions/issues"
+[tool.poetry.dependencies]
+python = ">=3.10, <3.12"
```
Comment (TheSuperiorStanislav) [inline pyproject.toml:27]: Please leave a comment why

File: tasks.py [hunk 10]
```diff
@@ -1,24 +1,42 @@
+import saritasa_invocations
 from invoke import Collection
 
-from provision import celery, ci, django, docker, git, linters, project, tests
+from provision import ci, project
```
Comment (TheSuperiorStanislav) [inline tasks.py:4]:
```suggestion
import provision
```

File: .github/workflows/checks.yml (hunk 3 above)
Comment (yalef) [reply .github/workflows/checks.yml:16]: We're using flake8-django for linting, it doesn't support python 3.12 yet until PR doesn't merged https://github.com/rocioar/flake8-django/pull/134

File: pyproject.toml (hunk 6 above)
Comment (yalef) [reply pyproject.toml:1]: We actually don't have homepage, but repository link might be useful. I'll add `documentation` also

Comment (TheSuperiorStanislav) [reply pyproject.toml:1]: What about docs page?

Comment (TheSuperiorStanislav) [reply pyproject.toml:1]: https://django-import-export-extensions.readthedocs.io/en/latest/

**Comment by yalef** [comment]
> Please, check if we can get rid of `import_export_extensions/__init__.py` file content

Only `__version__` var is used for sphinx config file, but it looks like it is optional and we can remove it

File: pyproject.toml (hunk 9 above)
Comment (yalef) [reply pyproject.toml]: Hm, actually we support 3.12, but we can't run linter checks for it due issue I've described above. I'll extend version to `^3.10`

**Review by OttoAndrey — APPROVED** [review]

File: Makefile
[the remark this answers is not in the document]
Comment (yalef) [reply Makefile:1]: Yes, I thought about it too. Let's do it in separate PR

File: tasks.py [hunk 11]
```diff
@@ -1,24 +1,42 @@
...
+    provision.ci,
+    saritasa_invocations.django,
+    saritasa_invocations.docker,
+    provision.project,
+    saritasa_invocations.pytest,
+    saritasa_invocations.poetry,
+    saritasa_invocations.git,
+    saritasa_invocations.pre_commit,
+    saritasa_invocations.mypy,
+    saritasa_invocations.python,
```
Comment (TheSuperiorStanislav) [inline tasks.py:17]:
```suggestion
    provision.project,
    provision.ci,
    saritasa_invocations.celery,
    saritasa_invocations.django,
    saritasa_invocations.docker,
    saritasa_invocations.pytest,
    saritasa_invocations.poetry,
    saritasa_invocations.git,
    saritasa_invocations.pre_commit,
    saritasa_invocations.mypy,
    saritasa_invocations.python,
```

File: pyproject.toml [hunk 12]
```diff
@@ -1,44 +1,148 @@
...
 classifiers = [
     "Development Status :: 3 - Alpha",
     "Intended Audience :: Developers",
-    "License :: OSI Approved :: MIT License",
     "Natural Language :: English",
     "Framework :: Django",
     "Framework :: Django :: 3.2",
     "Framework :: Django :: 4.0",
     "Framework :: Django :: 4.1",
     "Framework :: Django :: 4.2",
```
Comment (TheSuperiorStanislav) [inline pyproject.toml:34]: Please add django 5.0

Comment (TheSuperiorStanislav) [reply pyproject.toml:34]: Or better check if poetry could set it on build. Use this to read whl files https://filext.com/file-extension/WHL
image

**Pull request merged**